contracts/governance/utils/VotesOverridable.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Checkpoints} from "../../utils/structs/Checkpoints.sol";
+import {Votes} from "./Votes.sol";
+import {SafeCast} from "../../utils/math/SafeCast.sol";
+
+/**
+ * @dev Extension of {Votes} that adds support for checkpointed delegations and balances. This is required
+ * to use the `GovernorOverrideDelegateVote` extension.
+ */
+abstract contract VotesOverridable is Votes {
+    using Checkpoints for Checkpoints.Trace208;
+
+    error VotesOverridableFutureLookup(uint256 timepoint, uint256 currentTimepoint);
+
+    mapping(address delegatee => Checkpoints.Trace208) private _delegateCheckpoints;
+    mapping(address account => Checkpoints.Trace208) private _balanceOfCheckpoints;
+
+    function _delegate(address account, address delegatee) internal virtual override {
+        address oldDelegate = delegates(account);
+
+        _delegateCheckpoints[account].push(clock(), uint160(delegatee));
+
+        emit DelegateChanged(account, oldDelegate, delegatee);
+        _moveDelegateVotes(oldDelegate, delegatee, _getVotingUnits(account));
+    }
```

Comment:
This doesn't call super._delegate.

My understanding is that this completelly bypasses the storage of `Votes.sol`, and calling super would cause Vote's storage to be used.

This creates several issues though:
- any other extention that override `_delegate` and calls super would be dependant on the linearisation order.
- storage from Votes is still allocated, and is accessible through `Votes._xxx` functions.

Reply:
Agree with this. I changed it to call super.